datadog-sidecar/src/service/telemetry/ffi.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+#![allow(clippy::missing_safety_doc)]
+
+use crate::service::{InstanceId, QueueId};
+use ddtelemetry::data::{Log as DDLog, LogLevel as DDLogLevel};
+use ddtelemetry::worker::{LogIdentifier, TelemetryActions};
+use std::collections::hash_map::DefaultHasher;
+use std::ffi::CStr;
+use std::hash::{Hash, Hasher};
+use std::ptr::NonNull;
+use tokio::sync::mpsc;
+use tracing::{error, warn};
+
+use super::{get_telemetry_action_sender, InternalTelemetryActions};
+
+#[repr(C)]
+pub enum FfiError {
+    Ok = 0,
+    PointerNull = 1,
+    OperationFailed = 2,
+    InvalidUtf8 = 3,
+    QueueFull = 4,
+}
+
+#[repr(C)]
+pub enum CLogLevel {
+    Error = 1,
+    Warn = 2,
+    Debug = 3,
+}
+
+impl From<CLogLevel> for DDLogLevel {
+    fn from(level: CLogLevel) -> Self {
+        match level {
+            CLogLevel::Error => DDLogLevel::Error,
+            CLogLevel::Warn => DDLogLevel::Warn,
+            CLogLevel::Debug => DDLogLevel::Debug,
+        }
+    }
+}
```

Comment:
No need to repeat it here, it's already #[repr(C)] in ddtelemetry crate, just use it directly.